xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
A couple of reasons: for one, not all of our completed "processes" are actually the result of running a subprocess, and also, we need something that wraps the result of a run (we need a way to create this, and it's not clear from the documentation that `subprocess.CompletedProcess` is intended to be invoked directly, or that it is easy to create such a thing with the proper attributes, other than getting it from `run`, which we don't use).

I think this isn't the right place for `CompletedProcess` to live (maybe `proc.py`), and I think there is probably more information we want to have available from those objects, but I don't think we want to subclass `subprocess.CompletedProcess`, unless I'm missing something.